modules/homepage/src/main/resources/SLING-INF/content/libs/lfs/Homepage/html.GET.html

```diff
@@ -20,7 +20,9 @@
     <script>
       // Redirect to /login if the user is not logged in
       if (window.Sling.getSessionInfo() === null || window.Sling.getSessionInfo().userID === 'anonymous') {
-        window.location = "/login.html";
+        window.location = "/login.html" +
+           // Provide the `resource` to redirect to after successful login
+           ((window.location.pathname.length > 1) ? ("?resource=" + encodeURI(window.location.pathname)) : "");
```

Comment:
```suggestion
           ((window.location.pathname.length > 1) ? ("?resource=" + encodeURIComponent(window.location.pathname)) : "");
```

`encodeURI` doesn't escape query chars like `&`, `?` and `#`.

Reply:
Actually, I think if we should preserve the query string and hash as well.
```suggestion
           ((window.location.pathname.length > 1) ? ("?resource=" + encodeURIComponent(window.location.pathname + window.location.search + window.location.hash)) : "");
```